pkg/operator/encryption/state_controller.go

```diff
@@ -0,0 +1,197 @@
+package encryption
+
+import (
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const stateWorkKey = "key"
+
+// encryptionStateController is responsible for creating a single secret in
+// openshift-config-managed with the name destName.  This single secret
+// contains the complete EncryptionConfiguration that is consumed by the API
+// server that is performing the encryption.  Thus this secret represents
+// the current state of all resources in encryptedGRs.  Every encryption key
+// that matches encryptionSecretSelector is included in this final secret.
+// This secret is synced into targetNamespace at a static location.  This
+// indirection allows the cluster to recover from the deletion of targetNamespace.
+// See getResourceConfigs for details on how the raw state of all keys
+// is converted into a single encryption config.  The logic for determining
+// the current write key is of special interest.
+type encryptionStateController struct {
+	queue              workqueue.RateLimitingInterface
+	eventRecorder      events.Recorder
+	preRunCachesSynced []cache.InformerSynced
+
+	encryptedGRs             map[schema.GroupResource]bool
+	destName                 string
```

Comment:
what is destName?

Reply:
the name of the secret that contains `EncryptionConfiguration` in `openshift-config-managed` ns

Reply:
follow-up: ecnryptionSecretName